src/main/java/io/jenkins/plugins/folderauth/listeners/FolderListener.java

```diff
@@ -0,0 +1,39 @@
+package io.jenkins.plugins.folderauth.listeners;
+
+import com.cloudbees.hudson.plugins.folder.AbstractFolder;
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.listeners.ItemListener;
+import hudson.security.AuthorizationStrategy;
+import io.jenkins.plugins.folderauth.FolderBasedAuthorizationStrategy;
+import jenkins.model.Jenkins;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * Listens for changes to {@link AbstractFolder}s in Jenkins and makes sure that the configuration of
+ * {@link FolderBasedAuthorizationStrategy} remains the same.
+ * <p>
+ * Does not do anything if {@link Jenkins#getAuthorizationStrategy} is not {@link FolderBasedAuthorizationStrategy}.
+ */
+@Extension
+@Restricted(NoExternalUse.class)
+public class FolderListener extends ItemListener {
+    @Override
+    public void onDeleted(Item item) {
+        AuthorizationStrategy a = Jenkins.get().getAuthorizationStrategy();
```

Comment:
If #19 gets through, this pull request would have to be re-written. Race conditions would not be possible because we will obtain the lock and prevent other threads from messing. Please also note that setting the authorization strategy calls   `Jenkins.saveQuietly()`.